saml2/saml2-service-provider/src/main/java/org/springframework/security/saml2/core/Saml2X509Credential.java

```diff
@@ -35,7 +37,9 @@
  * @author Josh Cummings
  * @since 5.4
  */
-public final class Saml2X509Credential {
+public final class Saml2X509Credential implements Serializable {
+
+	private static final long serialVersionUID = SpringSecurityCoreVersion.SERIAL_VERSION_UID;
```

Comment:
Since this didn't have a serial version UID declared before 6.3 (when Spring Security started ensuring backward compatibility with Java serialization), it needs to be the JVM-calculated value.

Reply:
This is the same for the other classes as well